docs/shared/CONTRIBUTING.md

```diff
@@ -0,0 +1,58 @@
+We have [good first issues][good-first-issue] for new contributors and [help
+wanted][help-wanted] issues for our other contributors.
+
+- `good first issue` has extra information to help you make your first contribution.
+- `help wanted` are issues suitable for someone who isn't a core maintainer.
+
+Maintainers will do their best regularly make new issues for you to solve and
+then help out as you work on them. 💖
+
+# Philosophy
+
+PRs are most welcome!
+
+- If there isn't an issue for your PR, please make an issue first and explain
+  the problem or motivation for the change you are proposing. When the solution
+  isn't straightforward, for example "Implement missing command X", then also
+  outline your proposed solution. Your PR will go smoother if the solution is
+  agreed upon before you've spent a lot of time implementing it.
+  - It's OK to submit a PR directly for problems such as misspellings or other
+    things where the motivation/problem is unambiguous.
```

Comment:
I like that clarification. I recently contributed a one-word spelling fix to a repo, and was super anxious that they would close the PR because I didn't first file an issue.